lib/ContractInteract/BrandedToken.js

```diff
@@ -0,0 +1,393 @@
+   */
+  static deployRawTx(
+    web3,
+    valueToken,
+    symbol,
+    name,
+    decimals,
+    conversionRate,
+    conversionRateDecimals,
+    organization,
+  ) {
+    if (!(web3 instanceof Web3)) {
+      throw new TypeError(
+        `Mandatory Parameter 'web3' is missing or invalid: ${web3}`,
+      );
+    }
+    if (!Web3.utils.isAddress(valueToken)) {
+      throw new TypeError(`Invalid valueToken address: ${valueToken}.`);
+    }
+    if (!Web3.utils.isAddress(organization)) {
+      throw new TypeError(`Invalid organization address: ${organization}.`);
+    }
+    if (!(conversionRate > 0)) {
+      throw new TypeError(`Invalid conversion rate: ${conversionRate}. It should be more than zero`);
+    }
+    if (!(conversionRateDecimals < 5)) {
+      throw new TypeError(`Invalid conversion rate decimal: ${conversionRateDecimals}. It should be less than 5`);
+    }
+
+    const abiBinProvider = new AbiBinProvider();
+    const bin = abiBinProvider.getBIN(ContractName);
+
+    const args = [
+      valueToken,
+      symbol,
+      name,
+      decimals,
+      conversionRate,
+      conversionRateDecimals,
+      organization,
+    ];
+
+    const contract = Contracts.getBrandedToken(web3, null, null);
+
+    return contract.deploy(
+      {
+        data: bin,
+        arguments: args,
+      },
+    );
+  }
+
+  /**
+   * This calculates branded tokens equivalent to given value tokens.
+   *
+   * @param valueTokens Amount of value token.
+   *
+   * @return {Promise<string>} Promise that resolves to amount of branded token.
+   */
+  convertToBrandedTokens(valueTokens) {
+    return this.contract.methods
+      .convertToBrandedTokens(valueTokens)
+      .call();
+  }
+
+  /**
+   * Request stake for given amount. Approval for stake amount to branded
+   * token is required before calling this method.
+   *
+   * @param stakeAmount Stake amount.
+   * @param txOptions Transaction options.
+   *
+   * @return {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  async requestStake(stakeAmount, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError(`Invalid transaction options: ${txOptions}.`);
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(
+        `Invalid from address ${txOptions.from} in transaction options.`,
+      );
+      return Promise.reject(err);
+    }
+
+    const mintedAmount = await this.convertToBrandedTokens(stakeAmount);
+    const tx = await this.requestStakeRawTx(stakeAmount, mintedAmount);
+    return Utils.sendTransaction(tx, txOptions);
+  }
+
+  /**
+   * Raw tx for request stake.
+   *
+   * @param stakeAmount Stake amount.
+   * @param mintAmount Amount that will be minted after staking.
+   *
+   * @return Promise<Object> Raw transaction object.
+   */
+  requestStakeRawTx(stakeAmount, mintAmount) {
+    return Promise.resolve(this.contract.methods.requestStake(stakeAmount, mintAmount));
```

Comment:
Why `Promise.resolve` and not simply:
```suggestion
    return this.contract.methods.requestStake(stakeAmount, mintAmount);
```